### DISCUSSION TITLE
How to use a Secret API Key?

### DISCUSSION THREAD

**Discussion by diegonc** [medusajs/medusa#10594] [Q&A] [answered]
I created a secret api key in the following section:
![image](https://github.com/user-attachments/assets/8234edd9-6c14-43ba-b09a-daabdc200f37)

Now how do I use it to make requests? Like listing products?

**Comment by jkuzmanovik** [comment]
In MedusaJS, the secret API key is used when you need to perform actions as an authenticated admin user directly from the frontend.

If your goal is to display products in your webshop (storefront), you should use a publishable API key instead. This ensures secure access while exposing only the necessary public data.

**Reply by diegonc** [reply]
Hmm, I would use the secret key from another backend service. Putting it in the frontend is not a good idea (everyone can see and use it).

**Comment by diegonc** [comment, marked as answer]
I just found it's decribed [here](https://docs.medusajs.com/api/admin#authentication)

An `Authorization: Basic {token}` header must be added to the request.

It's kinda a twisted way of using basic authentication...
